### PR TITLE
Use Faraday rather than Excon for Switchboard lookups

### PR DESCRIPTION
**Pull request by stevenwilkin**
We're using Faraday for Google Custom Search so be consistent

### CODE REVIEW

File: lib/switchboard.rb [hunk 1]
```diff
@@ -20,5 +14,13 @@ def lookup(id)
     def base_url
       ENV['SWITCHBOARD_BASE_URL']
     end
+
+    def connection
```
Comment (andrewgarner) [inline lib/switchboard.rb]: Why not use https://github.com/guidance-guarantee-programme/pension_guidance/blob/master/lib/http_connection_factory.rb ?

Comment (stevenwilkin) [inline lib/switchboard.rb]: I looked at that but didn't think of using it, good call!

**Comment by stevenwilkin** [comment]
Updated to use `HTTPConnectionFactory `

File: lib/switchboard.rb (hunk 1 above)
Comment (andrewgarner) [inline lib/switchboard.rb]: The approach we've taken with Search is to pull the connection from a registry - https://github.com/guidance-guarantee-programme/pension_guidance/blob/master/config/initializers/search.rb#L1-L2

**Comment by andrewgarner** [comment]
How about using the registry pattern as we have for Search? - https://github.com/guidance-guarantee-programme/pension_guidance/blob/master/config/initializers/search.rb#L1-L2

**Comment by stevenwilkin** [comment]
This is complicated a bit by using the presence of the environment variable as a feature toggle

**Comment by andrewgarner** [comment]
Where do we do that?

**Comment by stevenwilkin** [comment]
Just rebased this in light of [this recent bugsnag](https://bugsnag.com/hm-treasury-1/pension-guidance/errors/56129c2902c5eb75212ef2a3).

The presence of the env var is used as a toggle [here](https://github.com/guidance-guarantee-programme/pension_guidance/blob/use-faraday-for-switchboard/lib/switchboard.rb#L4) @andrewgarner.

**Comment by stevenwilkin** [comment]
Updated to use the registry

**Comment by andrewgarner** [comment]
:+1:

**Pull request merged**